lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -445,61 +406,139 @@ function Get-CharmServices {
             )
         }
     }
-
+    $s2dContextGen = @{
+        "generator" = (Get-Item "function:Get-S2DCSVContext").ScriptBlock
+        "relation" = "csv"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'csv')) {
+        $s2dContextGen['mandatory'] = $true
+    }
+    $freeRdpContextGen = @{
+        "generator" = (Get-Item "function:Get-FreeRDPContext").ScriptBlock
+        "relation" = "free-rdp"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'free-rdp')) {
+        $freeRdpContextGen['mandatory'] = $true
```

Comment:
Yes, totally agree with you on this as well. I just wanted to let the deployer aware whether his unit is successfully cluster or not. The difference in the juju status message is more than enough.